components/clp-mcp-server/clp_mcp_server/validator/validate.py

```
@@ -0,0 +1,55 @@
+
+from antlr4 import CommonTokenStream, InputStream
+from antlr4.error.ErrorListener import ErrorListener
+
+from .generated.KqlLexer import KqlLexer
+from .generated.KqlParser import KqlParser
+
+
+class KqlErrorListener(ErrorListener):
+    """Custom error listener to capture syntax errors during parsing."""
+
+    def __init__(self) -> None:
```

Ruff (0.14.3)

Missing docstring in `__init__`

(D107)